commit 3bf0626ce7112e79886ac26373cf00f7588945a9 (refs/changes/04/2704/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-10-03T20:33:10+02:00 (2 years ago)
    
    CAPI-538 test-ufds: delete customer test randomly failing with InvalidArgumentError

diff --git a/test/capi/customers.test.js b/test/capi/customers.test.js
index 6ba4ca4..44400fe 100644
--- a/test/capi/customers.test.js
+++ b/test/capi/customers.test.js
@@ -793,9 +793,40 @@ test('delete key', function (t) {
 
 test('delete customer', function (t) {
     CAPI.del('/customers/' + CUSTOMER.uuid, function (err, req, res) {
-        t.ifError(err);
-        t.equal(200, res.statusCode);
-        t.end();
+        if (!err) {
+            t.equal(200, res.statusCode);
+            t.end();
+            return;
+        }
+        if (err.name !== 'InvalidArgumentError') {
+            t.ifError(err);
+            t.end();
+            return;
+        }
+        // napi-ufds-watcher could add a dclocalconfig entry to our
+        // customer and we'll have an error trying to delete it w/o removing
+        // every child entry. Let's try to remove it:
+        var localCfgDn = util.format(
+            'dclocalconfig=%s, uuid=%s, ou=users, %s',
+            CONFIG.datacenter_name, CUSTOMER.uuid, SUFFIX);
+
+        helper.createClient(function (clientErr, client) {
+            t.ifError(clientErr);
+            t.ok(client);
+            client.del(localCfgDn, function (err1) {
+                t.ifError(err1);
+                client.unbind(function (err2) {
+                    t.ifError(err2);
+                    CAPI.del('/customers/' + CUSTOMER.uuid,
+                        function (err3, req3, res3) {
+                        t.ifError(err3);
+                        t.equal(200, res3.statusCode);
+                        t.end();
+                    });
+                });
+            });
+        });
+
     });
 });
 
